### PR TITLE
Merge from private/auth-master

### PR DESCRIPTION
**Pull request by XiangtianDai**
Features include

- Allows the app to handle continue URL natively.
- Allows the app to set language code, e.g., for sending password reset email.

### CODE REVIEW

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m [hunk 1]
```diff
@@ -118,15 +179,26 @@ - (void)testPasswordResetRequest {
   XCTAssert([_RPCIssuer.decodedRequest isKindOfClass:[NSDictionary class]]);
   XCTAssertEqualObjects(_RPCIssuer.decodedRequest[kEmailKey], kTestEmail);
   XCTAssertEqualObjects(_RPCIssuer.decodedRequest[kRequestTypeKey], kPasswordResetRequestTypeValue);
+   XCTAssertEqualObjects(_RPCIssuer.decodedRequest[kContinueURLKey], kContinueURL);
```
Comment (paulb777) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m]: uneven indent

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m [hunk 2]
```diff
@@ -144,6 +216,32 @@ - (void)testEmailVerificationRequest {
...
+  XCTAssertEqualObjects(_RPCIssuer.decodedRequest[kAndroidInstallAppKey],
+                        [NSNumber numberWithBool:YES]);
+  XCTAssertEqualObjects(_RPCIssuer.decodedRequest[kCanHandleCodeInAppKey],
+                        [NSNumber numberWithBool:YES]);
+}
+
+#pragma mark - Helpers
+
+/** @fn fakeActionCodeSettings
+    @brief Contructs and retuns a fake instance of @c FIRActionCodeSettings for testing.
```
Comment (paulb777) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m]: Constructs and returns

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m [hunk 3]
```diff
@@ -69,13 +70,47 @@
...
+static NSString *const kUnauthorizedDomainErrorMessage = @"ERROR_UNAUTHORIZED_DOMAIN";
 
 /** @var kInvalidRecipientEmailErrorMessage
     @brief This is the prefix for the error message the server responds with if the recipient email
         is invalid.
  */
 static NSString *const kInvalidRecipientEmailErrorMessage = @"INVALID_RECIPIENT_EMAIL";
 
+/** @var kInvalidContinueURIErrorMessage
+    @brief This is the error returne by the backend if the continue URL provided in the request is
```
Comment (paulb777) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: returned

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m [hunk 4]
```diff
@@ -317,4 +492,16 @@ - (void)testSuccessfulEmailVerificationResponse {
   XCTAssertEqualObjects(RPCResponse.OOBCode, kTestOOBCode);
 }
 
+#pragma mark - Helpers
+
+/** @fn fakeActionCodeSettings
+    @brief Contructs and retuns a fake instance of @c FIRActionCodeSettings for testing.
```
Comment (paulb777) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: Constructs and returns

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m [hunk 5]
```diff
@@ -317,4 +492,16 @@ - (void)testSuccessfulEmailVerificationResponse {
...
+    @return An instance of @c FIRActionCodeSettings for testing.
+ */
+- (FIRActionCodeSettings *)fakeActionCodeSettings {
+  FIRActionCodeSettings *actionCodeSettings = [[FIRActionCodeSettings alloc]init];
```
Comment (paulb777) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: Space after alloc]

**Comment by XiangtianDai** [comment]
@paulb777 Yes my bad. I believe all is fixed now.

File: Firebase/Auth/Source/RPCs/FIRGetOOBConfirmationCodeRequest.h [hunk 6]
```diff
@@ -16,9 +16,12 @@
 
 #import <Foundation/Foundation.h>
 
+
```
Comment (paulb777) [inline Firebase/Auth/Source/RPCs/FIRGetOOBConfirmationCodeRequest.h]: Extra blank line?

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m (hunk 1 above)
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m]: Done.

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m (hunk 2 above)
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeRequestTests.m]: Done.

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m (hunk 3 above)
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: Done.

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m (hunk 4 above)
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: Done.

File: Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m (hunk 5 above)
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetOOBConfirmationCodeResponseTests.m]: Done.

File: Firebase/Auth/Source/RPCs/FIRGetOOBConfirmationCodeRequest.h (hunk 6 above)
Comment (XiangtianDai) [inline Firebase/Auth/Source/RPCs/FIRGetOOBConfirmationCodeRequest.h]: Done.

**Pull request merged**